livelossplot/output_plugins/extrema_print.py

```diff
@@ -0,0 +1,63 @@
+from typing import Dict, List
+
+from livelossplot.main_logger import LogItem
+from livelossplot.main_logger import MainLogger
+from .base_output import BaseOutput
+
+
+class ExtremaPrint(BaseOutput):
+    def __init__(self, massage_template: str = '\t{:16} \t (min: {:8.3f}, max: {:8.3f}, cur: {:8.3f})'):
+        self.massage_template = massage_template
+        self.extrema_cache = {}
+
+    def send(self, logger: MainLogger):
+        log_groups = logger.grouped_log_history()
+        massages = self._create_massages(log_groups)
+        print('\n'.join(massages))
+
+    def _create_massages(self, log_groups: Dict[str, Dict[str, List[LogItem]]]) -> List[str]:
+        massages = []
+        for group_name, group_logs in log_groups.items():
+            massages.append(group_name)
+            for metric_name, metric_values in group_logs.items():
+                min_val, max_val, current_val = self.get_extrema(metric_name, metric_values)
+                msg = self.massage_template.format(
+                    metric_name,
+                    min_val,
+                    max_val,
+                    current_val)
+                massages.append(msg)
+        return massages
+
+    def get_extrema(self, metric_name: str, metric_values: List[LogItem], skip_cache: bool = False):
+        current_val = metric_values[-1].value
+        cache = self.extrema_cache.get(metric_name)
+        if not cache or skip_cache:
+            min_val = min(metric_values, key=lambda i: i.value).value
+            max_val = max(metric_values, key=lambda i: i.value).value
+            current_val = metric_values[-1].value
+            self.extrema_cache[metric_name] = {
+                'min': min_val,
+                'max': max_val,
+                'current': current_val,
+            }
+        else:
+            min_val = min(cache['min'], current_val)
+            max_val = max(cache['max'], current_val)
+            cache['min'] = min_val
+            cache['max'] = max_val
+            cache['current'] = current_val
+        return min_val, max_val, current_val
```

Comment:
Personally, I would prefer to avoid (unnamed) tuple import of a function. If there are 3 elements or so, it is easy to make a mistake: which one is first, which last, etc. Maybe `dict`?

Reply:
so self.extrema_cache[metric_name]?

Reply:
Method "update_extrema".  And in "create_message" use values directly from cache.